### ISSUE TITLE
WARN logging occurs on containers without IPs, flooding logs every minute

### ISSUE THREAD

**Issue by timvanderkooi** [DataDog/datadog-agent#3796]
**Describe what happened:**

Every minute, the agent will log something like:

2019-06-28 20:57:02 UTC | PROCESS | WARN | (pkg/util/docker/containers.go:223 in parseContainerNetworkAddresses) | Unable to parse IP:  for container: /twistlock_defender_18_11_128"

This started occurring after the latest update to 6.12. Since we have our agent set to WARN logging, this floods our logging every minute with these types of logs from all of containers without IPs.

**Describe what you expected:**

This logging should be INFO instead of WARN. No IPs is an expected part of our deployment since some containers do not require an IP.

**Steps to reproduce the issue:**

Create a container without an IP

**Additional environment details (Operating System, Cloud provider, etc):**

**Comment by DylanLovesCoffee** [comment]
hey @timvanderkooi, thanks for raising the issue. We're noticing that the logs are being populated for pause containers, since their network mode is `none` when coupled with K8s (since networking is delegated to an underlying network plugin). This was introduced in 6.12.0, and although the logs are harmless we'll work on this and update you here when a change is made.

**Comment by xlucas** [comment]
Quick update: a fix has been merged for this, it will be available in the next release.

**Comment by timvanderkooi** [comment]
@xlucas thanks! Appreciate it.

**Comment by davidmoshal** [comment]
Odd, Datadog itself doesn't log errors, but when the same server is also instrumented, eg with LogDNA, then LogDNA logs these errors from the datadog-agent:

```
Dec 10 22:15:01 1c714f9dd34a datadog-agent INFO UTC | PROCESS | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /frontend will be missing network info: %!s(<nil>)
Dec 10 22:15:01 1c714f9dd34a datadog-agent INFO UTC | PROCESS | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /engine will be missing network info: %!s(<nil>)
Dec 10 22:15:01 1c714f9dd34a datadog-agent INFO UTC | PROCESS | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /nginx will be missing network info: %!s(<nil>)
Dec 10 22:15:01 1c714f9dd34a datadog-agent INFO UTC | PROCESS | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /redis will be missing network info: %!s(<nil>)
Dec 10 22:15:08 1c714f9dd34a datadog-agent INFO UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /frontend will be missing network info: %!s(<nil>)
Dec 10 22:15:08 1c714f9dd34a datadog-agent INFO UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /engine will be missing network info: %!s(<nil>)
Dec 10 22:15:08 1c714f9dd34a datadog-agent INFO UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /nginx will be missing network info: %!s(<nil>)
Dec 10 22:15:08 1c714f9dd34a datadog-agent INFO UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /redis will be missing network info: %!s(<nil>)
```

**Comment by dockerpoa** [comment]
I have the same error:

`2020-01-28 12:45:27 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:45:33 UTC | CORE | WARN | (pkg/collector/python/datadog_agent.go:118 in LogMessage) | kubelet:d884b5186b651429 | (kubelet.py:317) | GET on kubelet s `/stats/summary` failed: 403 Client Error: Forbidden for url: https://192.168.0.94:10250/stats/summary/?verbose=True
2020-01-28 12:45:42 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:45:42 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:45:59 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:45:59 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:46:05 UTC | CORE | WARN | (pkg/collector/python/datadog_agent.go:118 in LogMessage) | kubelet:d884b5186b651429 | (kubelet.py:317) | GET on kubelet s `/stats/summary` failed: 403 Client Error: Forbidden for url: https://192.168.0.94:10250/stats/summary/?verbose=True
2020-01-28 12:46:12 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:46:12 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:46:17 UTC | CORE | WARN | (pkg/collector/python/datadog_agent.go:118 in LogMessage) | kubelet:d884b5186b651429 | (kubelet.py:317) | GET on kubelet s `/stats/summary` failed: 403 Client Error: Forbidden for url: https://192.168.0.94:10250/stats/summary/?verbose=True
2020-01-28 12:46:29 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:46:29 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:46:44 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:46:44 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:46:51 UTC | CORE | WARN | (pkg/collector/python/datadog_agent.go:118 in LogMessage) | kubelet:d884b5186b651429 | (kubelet.py:317) | GET on kubelet s `/stats/summary` failed: 403 Client Error: Forbidden for url: https://192.168.0.94:10250/stats/summary/?verbose=True
2020-01-28 12:47:00 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:47:00 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:47:19 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:47:19 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:47:20 UTC | PROCESS | INFO | (collector.go:105 in runCheck) | Finish container check #120 in 3.202090752s
2020-01-28 12:47:22 UTC | CORE | WARN | (pkg/collector/python/datadog_agent.go:118 in LogMessage) | kubelet:d884b5186b651429 | (kubelet.py:317) | GET on kubelet s `/stats/summary` failed: 403 Client Error: Forbidden for url: https://192.168.0.94:10250/stats/summary/?verbose=True
2020-01-28 12:47:34 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:47:34 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:47:41 UTC | CORE | WARN | (pkg/collector/python/datadog_agent.go:118 in LogMessage) | kubelet:d884b5186b651429 | (kubelet.py:317) | GET on kubelet s `/stats/summary` failed: 403 Client Error: Forbidden for url: https://192.168.0.94:10250/stats/summary/?verbose=True
2020-01-28 12:47:46 UTC | CORE | INFO | (pkg/serializer/serializer.go:283 in SendMetadata) | Sent metadata payload, size (raw/compressed): 1113/466 bytes.
2020-01-28 12:47:46 UTC | CORE | INFO | (pkg/serializer/serializer.go:303 in SendJSONToV1Intake) | Sent processes metadata payload, size: 339 bytes.
2020-01-28 12:47:49 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:47:49 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:48:05 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:48:05 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:48:11 UTC | CORE | WARN | (pkg/collector/python/datadog_agent.go:118 in LogMessage) | kubelet:d884b5186b651429 | (kubelet.py:317) | GET on kubelet s `/stats/summary` failed: 403 Client Error: Forbidden for url: https://192.168.0.94:10250/stats/summary/?verbose=True
2020-01-28 12:48:21 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:48:21 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:48:36 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:48:36 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)
2020-01-28 12:48:42 UTC | CORE | WARN | (pkg/collector/python/datadog_agent.go:118 in LogMessage) | kubelet:d884b5186b651429 | (kubelet.py:317) | GET on kubelet s `/stats/summary` failed: 403 Client Error: Forbidden for url: https://192.168.0.94:10250/stats/summary/?verbose=True
2020-01-28 12:48:54 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /etcd1 will be missing network info: %!s(<nil>)
2020-01-28 12:48:54 UTC | CORE | ERROR | (pkg/util/docker/containers.go:110 in ListContainers) | Failed to get host IPs. Container /calico-rr will be missing network info: %!s(<nil>)`

I'm using k8s and deployment datadog-agent:7.16.1

**Comment by rlarivieredevo** [comment]
I have the same error and did some tests. I noticed that the problem occurs with Datadog 7 (installed using helm chart version 1.39.5). When I switched to Datadog 6.14.0, it was working fine.

**Comment by rpf3** [comment]
I was getting this error when running agent version 6.12.0-jmx but fixed it by upgrading to 7.19.2-jmx.
